spec/services/encryption/multi_region_kms_client_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe Encryption::MultiRegionKMSClient do
+  before do
+    stub_mapped_aws_kms_client(
+      'a' * 3000 => 'kms1',
+    )
+
+    allow(FeatureManagement).to receive(:use_kms?).and_return(kms_enabled)
+  end
+
+  let(:plaintext) { 'a' * 3000 }
+  let(:encryption_context) { { 'context' => 'attribute-bundle', 'user_id' => '123-abc-456-def' } }
+
+  kms_regions = JSON.parse(Figaro.env.aws_kms_regions)
+
+  let(:regionalized_kms_ciphertext) do
+    region_hash = {}
+    kms_regions.each do |r|
+      region_hash[r] = 'kms1'
+    end
+    { regions: region_hash }.to_json
+  end
+
+  let(:legacy_kms_ciphertext) do
+    'kms1'
+  end
+
+  let(:kms_enabled) { true }
+
+  aws_key_id = Figaro.env.aws_kms_key_id
+
+  describe '#encrypt' do
+    context 'with multi region enabled' do
```

Comment:
So the point of the multi-region KMS is that there is some redundancy, right? Can we get a spec that has the first KMS region fail, so we can verify that the multi-region client fails over to a secondary region, or at least a spec that makes sure that we can decrypt from multiple regions?

Reply:
Yeah this is on the right track! I meant more if we make an API request to a region and it fails, ex a network connect error due to a region being down. If I'm reading the tests as-is, these check for available configurations, which means that if an AWS region goes down, we'd need to do a deploy with new configs to fail over to an active region?

I added some similar auto-failover logic to our Telephony gem, I'm looking for some sort of `rescue` that catches network errors and fails over that way:

https://github.com/18F/identity-telephony/pull/25/files#diff-b9585e3dc0723db8732b54e133ab445aR28-R33

Reply:
Actually let's push failover on to a separate PR. This is good scope for now!